packages/rest/src/writer.ts

```diff
@@ -19,14 +19,18 @@ export function writeResultToResponse(
   result: OperationRetval,
 ): void {
   if (result) {
-    if (typeof result === 'object') {
-      // TODO(ritch) remove this, should be configurable
-      // See https://github.com/strongloop/loopback-next/issues/436
-      response.setHeader('Content-Type', 'application/json');
-      // TODO(bajtos) handle errors - JSON.stringify can throw
-      result = JSON.stringify(result);
-    } else if (typeof result === 'string') {
-      response.setHeader('Content-Type', 'text/plain');
+    switch (typeof result) {
+      case 'object':
+        // TODO(ritch) remove this, should be configurable
+        // See https://github.com/strongloop/loopback-next/issues/436
+        response.setHeader('Content-Type', 'application/json');
+        // TODO(bajtos) handle errors - JSON.stringify can throw
+        result = JSON.stringify(result);
+        break;
+      default:
+        response.setHeader('Content-Type', 'text/plain');
+        result = result.toString();
```

Comment:
We should follow Express conventions - see https://github.com/expressjs/express/blob/master/lib/response.js.
For example, `number` and `boolean` should be sent as `application/json`.

Reply:
I even think everything should be `JSON.stringified` by default. This way:
```
[1, 'a'] => [1, "a"]
{a: 1} => {"a": 1}
null => null
1 => 1
true => true
false => false
'string' => "string"
date => "2017-10-24T22:04:19.094Z"
```

Reply:
@raymondfeng I quickly attempted to do this just now, but it causes over 30 test failures. If this is something we want, I'll have to come back to it later.

That being said, I'd rather a 2nd attempt at this be focused more on making a robust and configurable writer component rather than the current implementation.